src/main/scala/org/broadinstitute/dsde/workbench/sam/service/UserService.scala

```diff
@@ -158,17 +158,10 @@ class UserService(val directoryDAO: DirectoryDAO, val cloudExtensions: CloudExte
     } yield status
   }
 
-  def getUserStatusInfo(userId: WorkbenchUserId, samRequestContext: SamRequestContext): IO[Option[UserStatusInfo]] =
-    directoryDAO.loadUser(userId, samRequestContext).flatMap {
-      case Some(user) =>
-        for {
-          adminEnabled <- directoryDAO.isEnabled(user.id, samRequestContext)
-        } yield {
-          val tosStatus = tosService.isTermsOfServiceStatusAcceptable(user)
-          Some(UserStatusInfo(user.id.value, user.email.value, tosStatus && adminEnabled, adminEnabled))
-        }
-      case None => IO.pure(None)
-    }
+  def getUserStatusInfo(user: SamUser, samRequestContext: SamRequestContext): IO[UserStatusInfo] = {
+    val tosStatus = tosService.isTermsOfServiceStatusAcceptable(user)
+    IO.pure(UserStatusInfo(user.id.value, user.email.value, tosStatus && user.enabled, user.enabled))
+  }
```

Comment:
This is the perf change. Instead of accepting an id then loading the user _again_ and calling `directoryDAO.isEnabled` which contains information already included in the user, just take in the full user object and reformat it to the expected type with the right logic around ToS.

Reply:
I couldn't figure out a good way to do this because it depends on biz logic in tosService.